plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergMetadata.java

```diff
@@ -492,7 +493,7 @@ public ConnectorOutputTableHandle beginCreateTable(ConnectorSession session, Con
     }
 
     @Override
-    public ConnectorInsertTableHandle beginInsert(ConnectorSession session, ConnectorTableHandle tableHandle)
+    public ConnectorInsertTableHandle beginInsert(ConnectorSession session, ConnectorTableHandle tableHandle, List<ColumnHandle> columns, RetryMode retryMode)
```

Comment:
For these metadata methods that take a RetryMode but it's ignored, should we be validating that it's `NO_RETRIES`?